sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -17,40 +17,256 @@ namespace Azure.AI.OpenAI
+        public OpenAIClient(Uri endpoint, string deploymentId, AzureKeyCredential credential) : this(endpoint, deploymentId, credential, new OpenAIClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of OpenAIClient. </summary>
+        /// <param name="endpoint">
+        /// Supported Cognitive Services endpoints (protocol and hostname, for example:
+        /// https://westus.api.cognitive.microsoft.com).
+        /// </param>
+        /// <param name="deploymentId"> The deployment id for operations initiated using this client instance </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public OpenAIClient(Uri endpoint, string deploymentId, TokenCredential credential) : this(endpoint, deploymentId, credential, new OpenAIClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of OpenAIClient. </summary>
+        /// <param name="endpoint">
+        /// Supported Cognitive Services endpoints (protocol and hostname, for example:
+        /// https://westus.api.cognitive.microsoft.com).
+        /// </param>
+        /// <param name="deploymentId"> The deployment id for operations initiated using this client instance </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public OpenAIClient(Uri endpoint, string deploymentId, AzureKeyCredential credential, OpenAIClientOptions options) : this(endpoint, credential, options)
+        {
+            DeploymentId = deploymentId;
+        }
+
+        /// <summary> Initializes a new instance of OpenAIClient. </summary>
+        /// <param name="endpoint">
+        /// Supported Cognitive Services endpoints (protocol and hostname, for example:
+        /// https://westus.api.cognitive.microsoft.com).
+        /// </param>
+        /// <param name="deploymentId"> The deployment id for operations initiated using this client instance </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public OpenAIClient(Uri endpoint, string deploymentId, TokenCredential credential, OpenAIClientOptions options) : this(endpoint, credential, options)
+        {
+            DeploymentId = deploymentId;
```

Comment:
Would it be valid if a user passed a null `deploymentId`? If not, then I recommend adding a check like the ones that the generated OpenAIClient does for the `endpoint` and the `credential` (see Generated/OpenAIClient.cs, line 72). Something like this:
```csharp
Argument.AssertNotNull(deploymentId, nameof(deploymentId));
```

And then modify the comment to also include the `deploymentId`:
```csharp
/// <exception cref="ArgumentNullException"> <paramref name="endpoint"/>, <paramref name="deploymentId"/>, or <paramref name="credential"/> is null. </exception>
```